src/Service/Telemetry/TelemetryTracesHelper.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using OpenTelemetry.Trace;
+
+namespace Azure.DataApiBuilder.Service.Telemetry
+{
+    public static class TelemetryTracesHelper
+    {
+        /// <summary>
+        /// Activity source for Data API Builder telemetry.
+        /// </summary>
+        public static readonly ActivitySource DABActivitySource = new("DataApiBuilder");
+
+        /// <summary>
+        /// Tracks the start of a REST controller activity.
+        /// </summary>
+        /// <param name="activity">The activity instance.</param>
+        /// <param name="httpMethod">The HTTP method of the request (e.g., GET, POST).</param>
+        /// <param name="userAgent">The user agent string from the request.</param>
+        /// <param name="actionType">The type of action being performed (e.g. Read).</param>
+        /// <param name="httpURL">The URL of the request.</param>
+        /// <param name="queryString">The query string of the request, if any.</param>
+        /// <param name="userRole">The role of the user making the request.</param>
+        /// <param name="apiType">The type of API being used (e.g., REST, GraphQL).</param>
+        public static void TrackRestControllerActivityStarted(
+            this Activity activity,
+            string httpMethod,
+            string userAgent,
+            string actionType,
+            string httpURL,
+            string? queryString,
+            string? userRole,
+            string apiType)
+        {
+            if (activity.IsAllDataRequested)
+            {
+                activity.SetTag("http.method", httpMethod);
+                activity.SetTag("user-agent", userAgent);
+                activity.SetTag("action.type", actionType);
+                activity.SetTag("http.url", httpURL);
+                if (queryString is not null && queryString != string.Empty)
```

Comment:
could use `!string.IsNullOrEmpty(queryString)`

Reply:
Is this a default style for the repo?